### PR TITLE
Bug 1924416: Bump dependencies to Kubernetes 1.20.6 to mitigate CVE-2021-3121

### PR DESCRIPTION
**Pull request by elmiko**
Now that there is a release on the Kubernetes 1.20 branch, v1.20.6, which mitigates this vulnerability. We can mitigate this issue by bumping the dependencies to include the fixed version of the protobuf dependency (v1.3.2 as shown in vendor/modules.txt).

To create this patch, in the go.mod, I replaced all Kubernetes versions
with `v0.20.6`, replaced the main Kubernetes import with `v1.20.6`
and ran make vendor.

### CODE REVIEW

File: cluster-autoscaler/go.mod [hunk 1]
```diff
@@ -390,36 +390,36 @@ replace (
...
+	k8s.io/cli-runtime => k8s.io/cli-runtime v0.20.6
+	k8s.io/client-go => k8s.io/client-go v0.20.6
+	k8s.io/cloud-provider => k8s.io/cloud-provider v0.20.6
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.20.6
+	k8s.io/code-generator => k8s.io/code-generator v0.20.6
+	k8s.io/component-base => k8s.io/component-base v0.20.6
+	k8s.io/component-helpers => k8s.io/component-helpers v0.20.6
+	k8s.io/controller-manager => k8s.io/controller-manager v0.20.6
+	k8s.io/cri-api => k8s.io/cri-api v0.20.6
+	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.20.6
```
Comment (JoelSpeed) [inline cluster-autoscaler/go.mod:406]: What happens if you remove these lines now? You are replacing a version with the same version, so I think you can get rid of these. Same for the others below and the final K8s.io replacement

File: cluster-autoscaler/vendor/modules.txt [hunk 2]
```diff
@@ -254,7 +254,7 @@ github.com/go-ozzo/ozzo-validation
 github.com/go-ozzo/ozzo-validation/is
 # github.com/godbus/dbus/v5 v5.0.3 => github.com/godbus/dbus/v5 v5.0.3
 github.com/godbus/dbus/v5
-# github.com/gogo/protobuf v1.3.1 => github.com/gogo/protobuf v1.3.1
+# github.com/gogo/protobuf v1.3.2 => github.com/gogo/protobuf v1.3.1
```
Comment (JoelSpeed) [inline cluster-autoscaler/vendor/modules.txt:257]: We need to be on v1.3.2, need to remove or update the replacement statement in `go.mod` for this one

**Review by JoelSpeed — COMMENTED** [review]
/bugzilla refresh

File: cluster-autoscaler/go.mod (hunk 1 above)
Comment (elmiko) [reply cluster-autoscaler/go.mod:406]: should work, i'll give it a try

File: cluster-autoscaler/vendor/modules.txt (hunk 2 above)
Comment (elmiko) [reply cluster-autoscaler/vendor/modules.txt:257]: i'll update it

File: cluster-autoscaler/go.mod (hunk 1 above)
Comment (elmiko) [reply cluster-autoscaler/go.mod:406]: i tried removing these from the `replace` section and when i run `go mod tidy` i get this error

```
go: k8s.io/kubernetes@v1.20.6 requires
        k8s.io/api@v0.0.0: reading k8s.io/api/go.mod at revision v0.0.0: unknown revision v0.0.0
```

**Comment by elmiko** [comment]
updated
* set gogo/protobuf at v1.3.2

**Review by JoelSpeed — COMMENTED** [review]
/approve

Because this differs slightly from how we normally manage these dependencies, let's get another reviewer to check they're happy with this

**Comment by elmiko** [comment]
> Because this differs slightly from how we normally manage these dependencies, let's get another reviewer to check they're happy with this

sounds good. i figure since this change is to a historical version it gives us a little more latitude in terms of how we decide to maintain the branch.

**Review by alexander-demichev — APPROVED** [review]
/lgtm

**Comment by deads2k** [comment]
(Patch Manager)

CVE, medium bz, 0.7

**Pull request merged**